From 12dcc28ce0aefb9c2b8ea743c9fc96acd9cd18c3 Mon Sep 17 00:00:00 2001
From: Nicolas Geoffray <ngeoffray@google.com>
Date: Thu, 8 Oct 2020 20:27:22 +0100
Subject: [PATCH 27/27] [master] Handle nterp when precompiling with JIT
 zygote.

Test: test.py
Test: app startup with JIT zygote
Bug: 169773857
Change-Id: I3e68e5e0dd276fd7fe11e435be0ce906bdca2a6c
---
 runtime/jit/jit.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/runtime/jit/jit.cc b/runtime/jit/jit.cc
index 5d6e47f56a..6c963aef7c 100644
--- a/runtime/jit/jit.cc
+++ b/runtime/jit/jit.cc
@@ -1339,6 +1339,7 @@ bool Jit::CompileMethodFromProfile(Thread* self,
   const void* entry_point = method->GetEntryPointFromQuickCompiledCode();
   if (class_linker->IsQuickToInterpreterBridge(entry_point) ||
       class_linker->IsQuickGenericJniStub(entry_point) ||
+      (entry_point == interpreter::GetNterpEntryPoint()) ||
       // We explicitly check for the stub. The trampoline is for methods backed by
       // a .oat file that has a compiled version of the method.
       (entry_point == GetQuickResolutionStub())) {
-- 
2.31.1

